sdc/tests/tests_perf/data_generator.py

```diff
@@ -16,14 +16,17 @@ def gen_str_data(data_num, data_length, input_data, data_width):
     return data
 
 
-def gen_data(data_num, data_length, input_data):
+def gen_data(data_num, data_length, input_data, typ):
```

Comment:
Are `input_data` and `typ` connected? If it's so, please use only one of these ones. There is a problem if `type(input_data)` doesn't match `typ`.